src/main/java/com/slisenko/education/html/PageDownloader.java

```diff
@@ -40,4 +45,49 @@ public void download(String link, String path) throws DownloadException {
             throw new DownloadException("Page loading error from link " + link + " to file " + path, e);
         }
     }
+
```

Comment:
please add timeout for page download, check
try this:
URL url = new URL(urlPath);
URLConnection con = url.openConnection();
con.setConnectTimeout(connectTimeout);
con.setReadTimeout(readTimeout);
InputStream in = con.getInputStream();